### PR TITLE
Blob Versioning and Jumbo Blob/Block

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**

### CODE REVIEW

**Comment by katmsft** [comment]
CI not passed. Please fix the CI before I can review it :).

File: Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp [hunk 1]
```diff
@@ -1021,6 +1021,7 @@ namespace azure { namespace storage {
             *snapshot.m_metadata = *resulting_metadata;
             snapshot.m_properties->copy_from_root(*properties);
             snapshot.m_properties->update_etag_and_last_modified(protocol::blob_response_parsers::parse_blob_properties(response));
+            properties->update_etag_and_last_modified(protocol::blob_response_parsers::parse_blob_properties(response));
```
Comment (katmsft) [inline Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp:1024]: What's it for?

Comment (JinmingHu-MSFT) [reply Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp:1024]: update the version id

Comment (JinmingHu-MSFT) [reply Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp:1024]: https://github.com/Azure/azure-storage-cpp/blob/c970b01af4918edc1cc6beb576837b38585abcbd/Microsoft.WindowsAzure.Storage/src/cloud_blob_shared.cpp#L35

update the version id

**Pull request merged**